modules/juce_audio_processors/processors/juce_AudioProcessorGraph.cpp

```diff
@@ -1262,6 +1262,17 @@ void AudioProcessorGraph::prepareToPlay (double sampleRate, int estimatedSamples
 
     clearRenderingSequence();
 
+    // Attention: Adobe megahack! 
+    // Force prepare all nodes here!
+    // Why are we doing this? 
+    // Because Premiere might:
+    //  - Trigger async preparation during render
+    //  - Report the isNonRealtime wrong
+    // This means that processing might start taking place before all nodes are
+    // prepped and ready, without a fallback wait/lock
+    for (auto* node : nodes)
+        node->prepare(sampleRate, estimatedSamplesPerBlock, this, getProcessingPrecision());

```

Comment:
Maybe we could also allow the regular preparation with `currentBlockSize` to take place by reseting the preparation state of each node to false? The `buildRenderingSequence` is using a scoped lock so it should be safe right?

```
for (auto* node : nodes)
{
    if (!node->isPrepared)
    {
        node->prepare(sampleRate, estimatedSamplesPerBlock, this, getProcessingPrecision());
        node->isPrepared = false; // so that we can allow the nodes to be reprepared during buildRenderingSequence if currentBlockSize differs from estimatedSamplesPerBlock
    }
}
```